src/App.test.jsx

```diff
@@ -0,0 +1,75 @@
+import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+import App from './App';
+import { MemoryRouter } from 'react-router-dom';
+import { UserProvider } from './context/UserContext';
+import { UserData } from './services/UserData';
+
+
+const server = setupServer(
+    rest.post(`${process.env.SUPABASE_API_URL}/auth/v1/token`, (req, res, ctx) => 
+    res(
+       ctx.json(UserData)
+    )
+  ),
+
+rest.get(`${process.env.SUPABASE_API_URL}/rest/v1/entries`, (req, res, ctx) => 
+  res(
+      ctx.json([
+          {
+            id: 1,
+            guest_id: '12345',
+            content: "Hello friends",
+            created_at: "2022-05-05T22:22:10.187801Z"
+          },
+          {
+            id: 2,
+            guest_id: '12345',
+            content: "Hello neighbor",
+            created_at: "2022-05-05T16:13:26.583814+00:00"
+          },
+      ])
+    )
+  )
+);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+describe('<App />', () => {
+  it('renders a list of user entries and has input for new user entries', async () => {
+      render(
+          <MemoryRouter>
+            <UserProvider>
+                <App />
+            </UserProvider>
+          </MemoryRouter>
+      );
+
+    const emailInput = screen.getByRole('textbox');
+    userEvent.type(emailInput, 'test_user@example.com');
+
+    const passwordInput = await screen.findByPlaceholderText(/password/i);
+    userEvent.type(passwordInput, 'random password');
+
+    const signInButton = await screen.findByRole('button', { name: /submit/i });
+    userEvent.click(signInButton);
+
+    waitForElementToBeRemoved(await screen.findByText(/loading/i));
+
+    const userEntry = await screen.findAllByRole('heading', { level: 3 });
+    expect(userEntry[0]).toBeInTheDocument();
+    expect(userEntry).toHaveLength(2);
+
+    const userEmail = await screen.findAllByRole('heading', { level: 4 });
+    expect(userEmail[0]).toBeInTheDocument();
+    expect(userEmail).toHaveLength(2);
+
+    const createdAt = await screen.findAllByRole('heading', { level: 5 });
+    expect(createdAt[0]).toBeInTheDocument();
+    expect(createdAt).toHaveLength(2);
```

Comment:
Instead of testing that the headings are present, let's also test that the actual text is on the page as well. We will want to check that 'hello neighbors' and 'hello friends' appear in the document.